client/Components/SingleCohort/Attendance.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+
+import { connect } from 'react-redux'
+
+import Table from '@material-ui/core/Table'
+import TableHead from '@material-ui/core/TableHead'
+import TableBody from '@material-ui/core/TableBody'
+import TableCell from '@material-ui/core/TableCell'
+import TableRow from '@material-ui/core/TableRow'
+
+import Button from '@material-ui/core/Button'
+
+const Attendance = props => {
+  const { students } = props
+
+  return (
+    <Table>
+      <EnhancedTableHead />
+
+      <TableBody>
+        {students.map(s => (
+          <StudentRow key={s.id} attributes={s} />
+        ))}
+      </TableBody>
+    </Table>
+  )
+}
+
+const EnhancedTableHead = props => {
+  return (
+    <TableHead>
+      <TableRow>
+        <TableCell>Student Name</TableCell>
+        <TableCell>Present</TableCell>
+        <TableCell>Excused</TableCell>
+        <TableCell>Edit</TableCell>
+      </TableRow>
+    </TableHead>
+  )
+}
+
+const StudentRow = props => {
+  const { attributes } = props
+  const { firstName, lastName, attendance } = attributes
+
+  return (
+    <TableRow>
+      <TableCell>
+        {firstName} {lastName}
+      </TableCell>
+
+      <TableCell>
+        {attendance ? (
+          `${attendance.isPresent ? 'Present' : 'Absent'}`
+        ) : (
+          <em>No record found</em>
+        )}
+      </TableCell>
+
+      <TableCell>
+        {attendance ? (
+          `${
+            attendance.isPresent
+              ? ''
+              : `${attendance.isExcused ? 'Excused' : 'Unexcused'}`
+          }`
+        ) : (
+          <em>No record found</em>
+        )}
+      </TableCell>
+
+      <TableCell>
+        <Button>Edit</Button>
+      </TableCell>
+    </TableRow>
+  )
+}
+
+const mapStateToProps = (state, props) => {
+  const { users } = state
+  const { cohortId, records } = props
+
+  const students = users
+    .filter(s => !s.isAdmin && s.cohortIds.find(id => id === cohortId))
```

Comment:
doesn't find return only the first instance of a criteria match?

Reply:
Yep, I'm using it to see if the cohortId of the SingleCohort component is in the user's cohortId's array.